src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -57,6 +61,7 @@ public SmugglerResult Execute()
                 var currentType = _source.GetNextType();
                 while (currentType != DatabaseItemType.None)
                 {
+                    _token.ThrowIfCancellationRequested();
```

Comment:
You need to move it to the ProcessDocuments, ProcessIndexes, etc. Those operations might take a lot of time and we want to cancel them.